### PR TITLE
Set HostRegexps for Service of type ExternalName

### LINKED ISSUE #1553: Non-matching requests are sent to route with Service of type ExternalName
**Describe the bug**
When an Ingress points to a Service of type `ExternalName`, skipper creates a route that matches every host. Requests that do not match a host/path configuration of any other route are then sent to that route.

**To Reproduce**
Ingress and Service definitions:
```yaml
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: my-external-service
  annotations:
    kubernetes.io/ingress.class: skipper
  labels:
    app: my-external-service
spec:
  rules:
    - host: my-external-service.domain.local
      http:
        paths:
          - backend:
              serviceName: my-external-service
              servicePort: 443
---
kind: Service
apiVersion: v1
metadata:
  name: my-external-service
  labels:
    app: my-external-service
spec:
  type: ExternalName
  externalName: google.com
  ports:
    - port: 443
      name: https
      targetPort: 443
```

The resulting route:
```
kube_default__my_external_service______google_com: *
  -> setRequestHeader("Host", "google.com")
  -> "https://google.com:443";
```

**Expected behavior**

skipper only proxies requests to the host `my-external-service.domain.local` to `google.com`.

**Observed behavior**

skipper proxies all non-matching requests, e.g. `http://127.0.0.1:9090/foo` to `google.com`.

### PR DESCRIPTION
**Pull request by wndhydrnt**
Fixes #1553

### CODE REVIEW

File: dataclients/kubernetes/kube_test.go [hunk 1]
```diff
@@ -614,7 +614,7 @@ func TestIngressData(t *testing.T) {
 			),
 		)},
 		map[string]string{
-			"kube_foo__qux______www_zalando_de": "https://www.zalando.de:443",
+			"kube_foo__qux__www_zalando_de____www_zalando_de": "https://www.zalando.de:443",
```
Comment (AlexanderYastrebov) [inline dataclients/kubernetes/kube_test.go:617]: The test is testing route id change but not the addition of `Host` predicate. 

I would be nice to modify test harness to check not only backend but rather whole route like, e.g.:
```go
map[string]string{
    "kube_foo__qux__www_zalando_de____www_zalando_de": "Host(/^www[.]zalando[.]de$/) -> setRequestHeader("Host", "www.zalando.de") -> "https://www.zalando.de:443""
```
Not sure if this is within the scope of this ticket though.

Comment (szuecs) [reply dataclients/kubernetes/kube_test.go:617]: out of scope, we have to change the tests similar to routegroup IMO

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
@wndhydrnt thanks, I never had time to investigate this!
related: https://github.com/zalando/skipper/issues/1157

File: dataclients/kubernetes/kube_test.go (hunk 1 above)
Comment (AlexanderYastrebov) [reply dataclients/kubernetes/kube_test.go:617]: There is `checkPrettyRoutes` that does the job.

**Comment by aermakov-zalando** [comment]
👍

**Pull request merged**